pkgs/applications/kde/marble.nix

```diff
@@ -14,4 +14,7 @@ mkDerivation {
     qtscript qtsvg qtquickcontrols qtwebkit shared-mime-info krunner kparts
     knewstuff gpsd
   ];
+  preConfigure = ''
+    cmakeFlags+=" -DINCLUDE_INSTALL_DIR=''${!outputDev}/include"
```

Comment:
I wasn't aware that the `placeholder` builtin had been added.
The nice thing about using `outputDev` is that it's valid regardless of whether or not a separate `dev` output exists.

Reply:
I agree, this should definitely use `outputDev`, otherwise it may silently fail.

Reply:
This package has control over the outputs it provides. And even if someone used `marble.overrideAttrs (super: { outputs = null; })`, removing dev output will definitely not cause a silent failure.

Reply:
I just want to point out that `outputs` can't be `null` or empty.
`marble.overrideAttrs (oldAttrs: { outputs = [ "out" ]; })` fails when trying to install to the temporary string provided by `placeholder "dev"`. In order to avoid it you would have to use: `marble.overrideAttrs (oldAttrs: { outputs = [ "out" ]; cmakeFlags = []; })`
I really don't see the advantage of that method .

Reply:
Placeholder variant is much more concise. Why would anyone in their right mind remove dev output?

Reply:
Concision is not a virtue above correctness.

Perhaps we could use `placeholder` with `assert (lib.elem "dev" outputs)`.

Reply:
It will be asserted by virtue of trying to install into the dev output. When the output does not exist, the installation will fail.

Reply:
I'm not saying it's a common thing but there is precedence for it in-tree:
[untrunc](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/video/untrunc/default.nix) overrides [libav_12](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/libav/default.nix) removing `dev`
[systemd-cryptsetup-generator](https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/systemd/cryptsetup-generator.nix) overrides [systemd](https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/systemd/default.nix) removing `dev`

Reply:
Failing during install, when we could fail during evaluation, is disrespectful to our users. Not failing is preferable to both.

At the moment, I am maintainer-by-default of this package. Because I'm responsible for providing support for it (what support I can), I would prefer it not fail if possible. If either of you or anyone else wants to take over maintenance of this package, I will defer entirely to that person.